backstage-plugin/plugins/open-dora/src/services/GroupDataService.test.ts

```diff
@@ -0,0 +1,121 @@
+import { MockConfigApi } from '@backstage/test-utils';
+import { rest } from 'msw';
+import { baseUrl, metricUrl } from '../../testing/mswHandlers';
+import { server } from '../setupTests';
+import { GroupDataService } from './GroupDataService';
+
+function createService() {
+  server.use(
+    rest.get(metricUrl, (req, res, ctx) => {
+      const params = req.url.searchParams;
+      const type = params.get('type');
+      const aggregation = params.get('aggregation');
+      const project = params.get('project');
+      const team = params.get('team');
+
+      return res(
+        ctx.json({
+          aggregation: aggregation || 'weekly',
+          dataPoints: [
+            {
+              key: `${project}_${team}_${aggregation}_${type}_first_key`,
+              value: 1.0,
+            },
+          ],
+        }),
+      );
+    }),
+  );
+  const mockConfig = new MockConfigApi({
+    'open-dora': { apiBaseUrl: baseUrl },
+  });
+
+  return new GroupDataService({ configApi: mockConfig });
+}
+
+describe('GroupDataService', () => {
+  it('should return data from the server', async () => {
+    const service = createService();
+
+    expect(
+      await service.retrieveMetricDataPoints({ type: 'df_count' }),
+    ).toEqual({
+      aggregation: 'weekly',
+      dataPoints: [{ key: 'null_null_null_df_count_first_key', value: 1 }],
+    });
+  });
+
+  it('should use provided details in the query parameters', async () => {
+    const service = createService();
+
+    expect(
+      await service.retrieveMetricDataPoints({
+        type: 'df_count',
+        aggregation: 'monthly',
+        project: 'project1',
+        team: 'team1',
+      }),
+    ).toEqual({
+      aggregation: 'monthly',
+      dataPoints: [
+        { key: 'project1_team1_monthly_df_count_first_key', value: 1 },
```

Comment:
I assume it wont cause any issues 1.0 (from line 22) and just 1

Reply:
Good spot, don't think so though since `1 === 1.0` is truth

Reply:
I can change it though just so the tests are clearer